lib/blockchain/chain.js

```diff
@@ -935,8 +935,10 @@ class Chain extends AsyncEmitter {
       const ns = await view.getNameState(this.db, nameHash);
 
       if (ns.isNull()) {
-        if (!covenant.isClaim() && !covenant.isOpen())
+        if (!covenant.isClaim() && !covenant.isOpen()) {
+          this.emit('CRITICAL ERROR', 'Unexpected null NameState.');
```

Comment:
I would probably use `abort` for the event name. smaller and more precise.

Reply:
i like it thanks